tests/test_units.py

```diff
@@ -185,3 +192,67 @@ def test_start_stop_backwards_compat():
     assert not 'started' in job
     assert not 'finished' in job
 
+def test_thread_raise():
+    let_thread_finish = threading.Event()
+    thread_preamble_done = threading.Event()
+    thread_caught_exception = None
+
+    def thread_target(accept_exceptions=False, block_exceptions=False):
+        if accept_exceptions:
+            brozzler.thread_accept_exceptions()
+        if block_exceptions:
+            brozzler.thread_block_exceptions()
+        thread_preamble_done.set()
+
+        try:
+            logging.info('waiting')
+            let_thread_finish.wait()
+        except Exception as e:
+            logging.info('caught exception %s', e)
+            nonlocal thread_caught_exception
+            thread_caught_exception = e
+        finally:
+            logging.info('finishing')
+            let_thread_finish.clear()
+            thread_preamble_done.clear()
+
+    # test that thread_raise does not raise exception in a thread that has not
+    # called thread_accept_exceptions
+    thread_caught_exception = None
+    th = threading.Thread(target=lambda: thread_target())
```

Comment:
I was going to recommend that this be just `threading.Thread(target=thread_target)`, but it's better and more consistent this way.